apps/charterafrica/src/lib/ecosystem/payload.js

```diff
@@ -0,0 +1,75 @@
+import * as Sentry from "@sentry/nextjs";
+
+import localize from "@/charterafrica/lib/ecosystem/utils";
+import api from "@/charterafrica/lib/payload";
+
+export async function updateOrCreate(collection, toCreate, locale) {
+  try {
+    const { docs } = await api.getCollection(collection, {
+      locale,
+      where: {
+        airtableId: { equals: toCreate.airtableId },
+      },
+    });
+    if (docs.length) {
+      const data = await api.updateCollection(
+        collection,
+        docs[0]?.id,
+        toCreate,
+        { locale }
+      );
+      return data;
+    }
+    const data = await api.createCollection(collection, toCreate, { locale });
+    return data;
+  } catch (e) {
+    Sentry.captureMessage(e.message);
+    return null;
+  }
+}
+
+export async function createCollection(collection, toCreate, { localized }) {
+  try {
+    const locales = localized ? ["en", "pt", "fr"] : ["en"];
+    const localizedData = localize(toCreate || {}, locales);
+    if (!localizedData) {
+      return null;
+    }
+    if (!localized) {
+      return updateOrCreate(collection, localizedData?.en);
+    }
+    const promises = Object.keys(localizedData).map((key) =>
+      updateOrCreate(collection, localizedData?.[key], key)
+    );
```

Comment:
```suggestion
    // All collections must have airtableId
    if (!toCreate?.airtableId) {
      return null;
    }
    const locales = localized ? ["en", "pt", "fr"] : ["en"];
    const dataPerLocale = localize(toCreate, locales);
    const promises = Object.keys(dataPerLocale).map((locale) =>
      updateOrCreate(collection, dataPerLocale.[locale], locale)
    );
```

Since we've only initialised `locales` based on `localized`, we no longer need to worry about it when calling `updateOrCreate`.